### PR TITLE
Disallow colon character in storage container names

### LINKED ISSUE #1564: Disallow ':' character in storage container names
File conventions disallow the colon (':') character, but jobs are allowed to come through with these in the name. We should replace all instances with dashes as per convention.

### PR DESCRIPTION
**Pull request by arinehouse**
Closes #1564

### CODE REVIEW

File: app/utils/storage-utils.spec.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ describe("StorageUtils", () => {
     let startAndEndRegex: RegExp;
 
     beforeEach(() => {
-        multipleCharRegex = (StorageUtils as any)._regexUnderscoresAndMultipleDashes;
+        multipleCharRegex = (StorageUtils as any)._regexInvalidCharacters;
```
Comment (timotheeguerin) [inline app/utils/storage-utils.spec.ts]: is this being used?

File: app/utils/storage-utils.spec.ts [hunk 2]
```diff
@@ -47,9 +49,8 @@ describe("StorageUtils", () => {
     });
 });
 
-function assertName(done: any, jobId: string, expected: string) {
+function assertName(jobId: string, expected: string) {
     StorageUtils.getSafeContainerName(jobId).then((containerName: string) => {
```
Comment (timotheeguerin) [inline app/utils/storage-utils.spec.ts]: why did you remove done? this is async no? should probably change to an async function all those tests instead of done anyway

Comment (arinehouse) [reply app/utils/storage-utils.spec.ts]: It's just calling `assertName` at the bottom of the file, which isn't async. Each `assertName` call was calling `done()` at the end, and the entire test would pass as soon as the first `assertName` completed, even if any others were failing.

File: app/utils/storage-utils.spec.ts (hunk 1 above)
Comment (arinehouse) [reply app/utils/storage-utils.spec.ts]: It's used in the 4th `it` block. I'm not sure exactly why though, maybe just to validate that the regex is what we expect, without testing the StorageUtils class? I can remove if it's unnecessary, I think it was a direct copy from the portal tests.
